subsystems/DiscordRPC/src/main/java/org/terasology/engine/subsystem/discordrpc/DiscordRPCThread.java

```diff
@@ -0,0 +1,284 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.subsystem.discordrpc;
+
+import com.jagrosh.discordipc.IPCClient;
+import com.jagrosh.discordipc.IPCListener;
+import com.jagrosh.discordipc.entities.RichPresence;
+import com.jagrosh.discordipc.exceptions.NoDiscordClientException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class DiscordRPCThread implements IPCListener, Runnable {
+    private static final Logger logger = LoggerFactory.getLogger(DiscordRPCThread.class);
+    private static final long DISCORD_APP_CLIENT_ID = 515274721080639504L;
```

Comment:
Is this a random number or something assigned to us in some way?

Reply:
This random represents the client ID of the Discord application. Discord client needs this ID to fetch information and assets about the application to display it to the rich presence.

Reply:
Yeah I think that might be associated with our registered app on Discord, which I originally set up. Nowadays the app is owned by a team with several of us having access 👍

Reply:
Is that _sensitive information_ (like an access token) or more of a global identifier?

Reply:
Just an identifier 👍 There's also a secret but I don't think it is needed for anything in this case.

Incidentally you should have access to it via https://discord.com/developers/teams/781638052501520417 although it looks like you might have gotten an invite and not accepted it yet (Nira is in)

Reply:
Yep as Cervator said it's an identifier so it's not sensitive information. And, about the secret, it's mostly used in OAuth2 applications which I think we don't need soon.

I agree with what Cervator said. And about the secret, it's mostly used in OAuth2 applications which I don't think we will need soon.